woocommerce/woocommerce-android
# Subscription review issues

### Description
This PR fixes several issues identified in the test charter review.

Fix #1 64e19b4b2ae745511931594c44dec84493b52ab6
> For a variable subscription, changing attributes removes the subscriptions section and replaces it with the price section. From the price screen, I can change prices. Those new prices are shown for the subscription variation in the web browser, but only the new sale price shows for the variation in the app (even if I go to the products list and pull to refresh or close/re-open the app).

Fix #2 6261f355755a0acb5c50ca4a41134bdad70906f5
> In a RTL language, the text on the subscription screen is still left aligned, rather than right aligned

Fix #3 18cb4223e5271ffc865b7a351517fff58c19ffe1
> With large font on the device and big price, the subscription numbers and price can overlap

Fix #4 e154ec06367ce43f6640f48fc65317b3ec7b8195
> Talkback: In the order details screen, talkback doesn’t read the price in an intuitive way. It says “Price” – “Subscription status” – “Time Period”, resulting in audio like “5 dollars active every month”.

Fix #5 eb42d088e11e502bb7c392fbc3a1b17bbbad2204
> The free trial display is able to interpret “0” as “No free trial,” but sign-up fee handles null and 0 differently. (Null = “No sign up fee” and 0 = “$0.00”) It isn’t incorrect that it shows “$0.00” rather than “No sign up fee,” but it would be nice if it did.

### Testing instructions
TC1
1. Open a variable subscription from the products list
2. Tap on variations
3. Tap on any variation
4. Tap on attributes and change the value
5. Check that the Price card is not show
6. Save the value and check that everything works as expected

TC2
1. Set a RTL language or Force RTL layout direction on developer options
2. Open a subscription product
3. Open the subscription details
4. Check that text is right-aligned

TC3
Increase the text font in a subscription with a big price and check that the subscription numbers and price don't overlap.

TC4
1. Activate Talkback
2. Open an order containing subscriptions
3. Tap on a subscription
4. Check that the subscription description is in the correct order

TC5
1. On the web-admin, set a subscription fee to $0.00 
2. Open the products list
3. Open the subscription with the fee of $0.00


### Images/gif

| TC1 | TC2 | TC5 |
| ------------- | ------------- | ------------- |
| <img width="300" src="https://user-images.githubusercontent.com/18119390/230188845-1e8664f5-7804-4ee7-9e24-40ca08bcb792.png" /> | <img width="300" src="https://user-images.githubusercontent.com/18119390/230188974-0b0ff90e-6489-4dfc-9e4a-f9b1c35b2172.png" />  | <img width="300" src="https://user-images.githubusercontent.com/18119390/230189096-02737c3c-11f1-402b-a926-b1153f76ab5f.png" />  |

TC3
<img width="280" alt="Screenshot 2023-04-05 at 13 42 09" src="https://user-images.githubusercontent.com/18119390/230190421-3c66c0f4-ad9c-410e-85aa-8c478293aacf.png">
